src/whir/utils.rs

```diff
@@ -91,6 +92,14 @@ pub trait DigestToUnitDeserialize<MerkleConfig: Config> {
     fn read_digest(&mut self) -> ProofResult<MerkleConfig::InnerDigest>;
 }
 
+pub trait HintSerialize {
```

Comment:
I think it would maybe be cleaner to have an hint trait for each type we want to serialize. This way at call site we can understand that we are doing the right hint addition